---
version: '3.5'
services:
  server:
    image: "python:3.7"
    ports:
      - "8080:8080"
    volumes:
      - "./:/opt/app"
      - "./.volumes/server_home:/root"
    working_dir: "/opt/app"
    entrypoint: '/bin/bash -c "pip install -e . && vending-machine run -h 0.0.0.0 -p 8080"'
    environment:
      FLASK_DEBUG: ${FLASK_DEBUG:-0}

  grafana:
    image: "grafana/grafana"
    ports:
      - "3000:3000"
    volumes:
    - "./.volumes/grafana/data:/var/lib/grafana"
    environment:
      GF_AUTH_ANONYMOUS_ENABLED: "true"
      GF_AUTH_BASIC_ENABLED: "false"

  prometheus:
    image: "prom/prometheus"
    ports:
    - "9090:9090"
    volumes:
      - "./.volumes/prometheus:/prometheus"
      - "./infra/prometheus/prometheus.yml:/etc/prometheus/prometheus.yml"

  node_exporter:
    image: "prom/node-exporter"
    ports:
      - "9100:9100"

  locust:
    build:
      context: infra/locust
      # Docker has a problem that this is relative to context, not the local filesystem.
      dockerfile: Dockerfile
    ports:
      - "8089:8089"
    volumes:
      - "./infra/locust:/opt/app"
    working_dir: "/opt/app"
    command: "locust -f generate_traffic.py --host http://server:8080"
